### ISSUE TITLE
Escaping/quoting of variable args in commands

### ISSUE THREAD

**Issue by jack1142** [tusharsadhwani/zxpy#8]
Proper handling of quoting in bash is always a pain point, it would be nice if the command substitution escaped special characters which is what `zx` is doing.
I'm not sure what would be the best way to do this while also ensuring that this doesn't hurt the usage where you *do* want it to not be escaped and I feel this could use more feedback from the community which this issue might help gather.

The current solution is using `shlex.quote()` but I think this makes it less straightforward.

Example code:
- `showargv.py`
```py
import sys

print(sys.argv)
```
- `simplescript.py`
```
#! /usr/bin/env zxpy

var = "test string"
output = ~f"python3.8 showargv.py {var}"

print(output)
```

This results in output:
```
['showargv.py', 'test', 'string']
```
While I think that more often than not you would want:
```
['showargv.py', 'test string']
```

**Comment by tusharsadhwani** [comment]
yeah this was discussed in a forum before, and yeah I think a simple `shlex.quote` in my implementation should fix most of the issues with quoting, shell injection etc.

I just haven't had enough time to test out the library, so thanks for creating the issues :smile:

I'll add a few tests in some time and see if quoting causes any other issues. Otherwise, this should be an easy addition.

**Comment by tusharsadhwani** [comment]
I've checked your example in release version 1.4.0, and uploaded it.

Feel free to reopen the issue if you have any questions :smile:
